core/scripts/cli/sponsor/create.js

```diff
@@ -0,0 +1,42 @@
+const inquirer = require("inquirer");
+const getDefaultAccount = require("../wallet/getDefaultAccount");
+
+const create = async (web3, artifacts, emp) => {
+  const ExpandedERC20 = artifacts.require("ExpandedERC20");
+  const { toWei, fromWei, toBN } = web3.utils;
+
+  const sponsorAddress = await getDefaultAccount(web3);
+
+  const scalingFactor = toBN(toWei("1"));
+  // TODO: Understand why we need a .rawValue in one case but not the other.
+  const totalPositionCollateral = toBN((await emp.totalPositionCollateral()).rawValue.toString());
+  const totalTokensOutstanding = toBN((await emp.totalTokensOutstanding()).toString());
+  if (totalTokensOutstanding.isZero()) {
+    console.log("Error: can't create initial position with this tool");
+  }
+  const gcr = totalPositionCollateral.mul(scalingFactor).divRound(totalTokensOutstanding);
+
+  const input = await inquirer.prompt({
+    message: "How many tokens to create?",
```

Comment:
you should constrain the input and make sure that input are numbers and are positive numbers. Similar to how I do so in `./core/scripts/cli/voting/commitVotes.js`:

```js
        const priceInput = await inquirer.prompt({
          type: "number",
          name: "price",
          default: 0.0,
          message: style.instruction("Enter a positive price. Invalid input will default to 0!"),
          validate: value => value >= 0 || "Price must be positive"
        });
```

More docs [here](https://github.com/SBoudrias/Inquirer.js#input---type-number)

Reply:
Checking >0 seems reasonable. The `toWei` function only allows strings to avoid precision errors, so I think taking the input as a string makes sense here as well. If someone really types in "five" here, they deserve a crash.

Reply:
haha. Why not just take in a number, and then toWei() it as you do?

Reply:
Because toWei only works with a string.

Reply:
could just toString() the input

Reply:
You'll still have precision loss. A string is safer given we allow down to 18 decimals.

Reply:
ok, then let's just check >0